workflow/controller/pod_cleanup.go

```diff
@@ -1,22 +1,147 @@
 package controller
 
 import (
+	"context"
+
 	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
 
 	"github.com/argoproj/argo-workflows/v3/workflow/common"
 
 	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 	"github.com/argoproj/argo-workflows/v3/workflow/controller/indexes"
 )
 
-func (woc *wfOperationCtx) queuePodsForCleanup() {
+func (woc *wfOperationCtx) isImmediateSkipQueues() bool {
+	return woc.rateLimiter != nil && woc.orig.Spec.PodGC != nil && woc.orig.Spec.PodGC.Strategy == wfv1.PodGCOnPodCompletion
+}
```

Comment:
I'm not sure if it's just me, but I'm a little confused at the name of the method - can you add a comment ahead of it to explain what it is?

Reply:
Sorry about being AWOL.  I just added a comment now.